test/integration/consul-container/test/upgrade/peers_http_test.go

```diff
@@ -0,0 +1,91 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/api"
+	libassert "github.com/hashicorp/consul/test/integration/consul-container/libs/assert"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+)
+
+const (
+	acceptingPeerName = "accepting-to-dialer"
+	dialingPeerName   = "dialing-to-acceptor"
+)
+
+func TestPeering_UpgradeToTarget_fromLatest(t *testing.T) {
```

Comment:
Not a dealbreaker, but I think it's helpful to add some context about why this test case exists or what it is testing in the comments.